src/NHibernate/Linq/Visitors/ExpressionParameterVisitor.cs

```diff
@@ -107,6 +102,17 @@ protected override Expression VisitMethodCall(MethodCallExpression expression)
 				return Expression.Call(null, expression.Method, query, arg);
 			}
 
+			if (_functionRegistry != null &&
+				_functionRegistry.TryGetGenerator(method, out var generator) &&
```

Comment:
This is potentially expensive operation. Could it be replaced with something lighter?

Reply:
Added an additional check for method name in order to avoid calling `TryGetGenerator` too often.

Reply:
Thanks.